packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -26,12 +27,29 @@ class IOSSimulators extends PollingDeviceDiscovery {
   IOSSimulators() : super('IOSSimulators');
 
   bool get supportsPlatform => Platform.isMacOS;
-  List<Device> pollingGetDevices() => IOSSimulator.getAttachedDevices();
+  List<Device> pollingGetDevices() => IOSSimulatorUtils.instance.getAttachedDevices();
+}
+
+class IOSSimulatorUtils {
+  /// Returns [IOSSimulatorUtils] active in the current app context (i.e. zone).
+  static IOSSimulatorUtils get instance => context[IOSSimulatorUtils] ?? (context[IOSSimulatorUtils] = new IOSSimulatorUtils());
```

Comment:
sgtm re: the globals cleanup, though I think there are some things we want to have available to much of the code, for convenience.